firewall/firewall.go

```diff
@@ -0,0 +1,49 @@
+package firewall
+
+import (
+	"time"
+
+	"github.com/bluele/gcache"
+	"github.com/modern-go/concurrent"
+)
+
+const WindowSize = 120 * time.Second
+const MaxStringsPerIp = 6
+
+var resourcesForIPCache = gcache.New(1000).Simple().Build()
+var whitelist = map[string]bool{
+	"51.210.0.171": true,
+}
+
+func IsIpAbusingResources(ip string, endpoint string) (bool, int) {
```

Comment:
The name sets an expectation that measuring some actual resource usage like network bandwidth is involved, I would suggest calling this in a way that makes it apparent we're checking rate limits.